### ISSUE TITLE
Distributed training 

### ISSUE THREAD

**Issue by DNJ-JINI** [keras-team/tf-keras#237]
I am using the following code to use distributed training with azure [GPU - 2 x NVIDIA Tesla V100]. So during the training process both the available  GPU  is working and the training time is also less compared with single GPU. 
My question is whether the data is divided among each GPU ( 16 samples per GPU or 32 samples per GPU since I have given the batch_size as 32). Also when batch_size is 64 it results in resource exhausted error. The model input shape is 256x256

batch_size = 32
dist_data_train = tf.data.Dataset.from_tensor_slices((X_train, y_train)).batch(batch_size)
dist_data_val = tf.data.Dataset.from_tensor_slices((X_test, y_test)).batch(batch_size)

strategy = tf.distribute.MirroredStrategy()
print('Number of devices: {}'.format(strategy.num_replicas_in_sync))

# Create a distributed model
with strategy.scope():
    model = get_model(img_size, num_classes)
    distributed_model = model
    distributed_model.compile(optimizer = tf.keras.optimizers.Adam(lr = 0.001), loss = ['sparse_categorical_crossentropy'], metrics=['accuracy'])

checkpoint = ModelCheckpoint(model_name,
                             monitor="val_loss",
                             mode="min",
                             save_best_only = True,
                             verbose=1)
early = EarlyStopping(monitor="val_loss", mode="min", patience=15)
callbacks=[early, checkpoint]
num_epochs= 200
#modelunet=distributed_model.fit(train_gen, epochs=num_epochs, validation_data=val_gen, callbacks=callbacks)
model=distributed_model.fit(dist_data_train, epochs=num_epochs, validation_data=dist_data_val, callbacks=callbacks)
